inlong-sort/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/internal/JdbcMultiBatchingOutputFormat.java

```diff
@@ -519,6 +512,9 @@ protected void attemptFlush() throws IOException {
                     if (!flushFlag && null != tableException) {
                         LOG.info("Put tableIdentifier:{} exception:{}",
                                 tableIdentifier, tableException.getMessage());
+                        outputMetrics(tableIdentifier, Long.valueOf(tableIdRecordList.size()),
+                                1L, true);
+                        dirtySinkHelper.invoke(record, DirtyType.RETRY_LOAD_ERROR, tableException);
```

Comment:
Dirty data sink should be after restart with exception when the 'isIgnoreTableException' is 'false'